ReactSkia/components/RSkComponentText.cpp

```diff
@@ -51,84 +52,135 @@ inline int getNumberOfLines(std::vector<LineMetrics>& metrics, int maxNumberOfLi
+                numberOfLines = getLines(metrics, frame.size.height - (2*rectStrokeWidth));
+            }
+            if (props.paragraphAttributes.maximumNumberOfLines) {
+                numberOfLines = getMaxNumberOfLines(metrics, props.paragraphAttributes.maximumNumberOfLines, frame.size.height);
+                if ((EllipsizeMode::Tail) == (props.paragraphAttributes.ellipsizeMode)) {
                     paraStyle.setEllipsis(u"\u2026");
                 }
-                paraStyle.setMaxLines(numberOfLines);
-                parent->paraBuilder->setParagraphStyle(paraStyle);
+            }
+
+            paraStyle.setMaxLines(numberOfLines);
+            parent->paraBuilder->setParagraphStyle(paraStyle);
+            if (numberOfLines) {
                 paragraph = parent->paraBuilder->Build();
-                paragraph->layout(frame.size.width);
+                paragraph->layout(frame.size.width - (2*rectStrokeWidth));
             }
-            yOffset = yPosOffset(data.attributedString, paragraph->getHeight(), frame);
-            paragraph->paint(canvas, frame.origin.x, frame.origin.y + yOffset);
+
+            if ((props.backgroundColor) || (props.textAttributes.lineHeight >= 0)) {
+                canvas->drawRect(SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height), convertTextColor(props.backgroundColor));
+                canvas->clipRect(SkRect::MakeXYWH(frame.origin.x,frame.origin.y,frame.size.width,frame.size.height));
+            }
+            yOffset = yPosOffset(data.attributedString, paragraph->getHeight(), frame.size.height - rectStrokeWidth);
+            paragraph->paint(canvas, frame.origin.x + rectStrokeWidth, frame.origin.y + yOffset + rectStrokeWidth);
+            RSkDrawUtils::drawBorder(canvas, frame, borderMetrics, props.backgroundColor, 1);
         }
     }else {
         /* If previously created builder is available,using it will append the text in builder*/
         /* If it reaches here,means there is an update in text.So create new paragraph builder*/
         if(nullptr != paraBuilder) {
             paraBuilder.reset();
         }
-
         paraBuilder = std::static_pointer_cast<ParagraphBuilder>(std::make_shared<ParagraphBuilderImpl>(paraStyle,data.layoutManager->collection_));
 
-        expectedAttachmentCount = data.layoutManager->buildParagraph(data.attributedString, props.paragraphAttributes, true, paraBuilder);
+        expectedAttachmentCount = data.layoutManager->buildParagraph(props.backgroundColor, data.attributedString, props.paragraphAttributes, true, paraBuilder);
         currentAttachmentCount = 0;
         auto paragraph = paraBuilder->Build();
 
         /* If the count is 0,means we have no fragment attachments.So paint right away*/
         if(!expectedAttachmentCount) {
             auto frame = getAbsoluteFrame();
-            paragraph->layout(frame.size.width);
+            paragraph->layout(frame.size.width - (2*rectStrokeWidth));
```

Comment:
since nested text and paragraph texts are having same code ; can we bring them under a function  and call it.. this is to avoid duplicating the same code at two places.

Reply:
Updated